File: implementation/scipamato/core/web/src/test/java/ch/difty/scipamato/core/web/common/CoreWebSessionFacadeTest.java
package ch.difty.scipamato.core.web.common;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Locale;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.mock.mockito.MockBean;

import ch.difty.scipamato.common.navigator.LongNavigator;
import ch.difty.scipamato.common.web.ScipamatoWebSessionFacade;
import ch.difty.scipamato.core.auth.Roles;
import ch.difty.scipamato.core.web.WicketTest;

class CoreWebSessionFacadeTest extends WicketTest {

    private final ScipamatoWebSessionFacade sessionFacade = new CoreWebSessionFacade();

    @Test
    void gettingLanguageCode_withBritishLocale_returnsBritishCode() {
        getTester()
            .getSession()
            .setLocale(new Locale("en_GB"));
        assertThat(sessionFacade.getLanguageCode()).isEqualTo("en_gb");
    }

    @Test
    void gettingLanguageCode_withFrenchLocale_returnsFrenchCode() {
        getTester()
            .getSession()
            .setLocale(new Locale("fr"));
        assertThat(sessionFacade.getLanguageCode()).isEqualTo("fr");
    }

    @Test
    void gettingPaperIdManager_returnsMock() {
        assertThat(sessionFacade.getPaperIdManager()).isInstanceOf(LongNavigator.class);
    }

    @Test
    void roleCheck() {
        assertThat(sessionFacade.hasAtLeastOneRoleOutOf(Roles.VIEWER, Roles.ADMIN)).isFalse();
        assertThat(sessionFacade.hasAtLeastOneRoleOutOf(Roles.USER)).isTrue();
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/common/CoreWebSessionFacadeTest.kt
package ch.difty.scipamato.core.web.common

import ch.difty.scipamato.common.navigator.LongNavigator
import ch.difty.scipamato.common.web.ScipamatoWebSessionFacade
import ch.difty.scipamato.core.auth.Roles
import ch.difty.scipamato.core.web.WicketTest
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeFalse
import org.amshove.kluent.shouldBeInstanceOf
import org.amshove.kluent.shouldBeTrue
import org.junit.jupiter.api.Test
import java.util.*

internal class CoreWebSessionFacadeTest : WicketTest() {

    private val sessionFacade: ScipamatoWebSessionFacade = CoreWebSessionFacade()

    @Test
    fun gettingLanguageCode_withBritishLocale_returnsBritishCode() {
        tester.session.locale = Locale.of("en_GB")
        sessionFacade.languageCode shouldBeEqualTo "en_gb"
    }

    @Test
    fun gettingLanguageCode_withFrenchLocale_returnsFrenchCode() {
        tester.session.locale = Locale.of("fr")
        sessionFacade.languageCode shouldBeEqualTo "fr"
    }

    @Test
    fun gettingPaperIdManager_returnsMock() {
        sessionFacade.paperIdManager shouldBeInstanceOf LongNavigator::class
    }

    @Test
    fun roleCheck() {
        sessionFacade.hasAtLeastOneRoleOutOf(Roles.VIEWER, Roles.ADMIN).shouldBeFalse()
        sessionFacade.hasAtLeastOneRoleOutOf(Roles.USER).shouldBeTrue()
    }
}
